invisible_cities/reco/hits_functions.py

```diff
@@ -63,3 +64,52 @@ def threshold_hits(hits : List[evm.Hit], th : float, on_corrected : bool=False)
                 x.E = es[i]
                 new_hits.append(x)
         return new_hits
+
+def number_of_hits_per_blob(hitc       : evm.HitCollection,
+                            blob_pos1  : Tuple[float, float, float],
+                            blob_pos2  : Tuple[float, float, float],
+                            blob_radius: float,
+                            energy_out : bool = True
+                            )-> Tuple[float, float, float, float]:
+    """
+    The functions computes the number of hits per blob and (if required) the
+    blob energy for a given hit collection taking the euclidian distance.
+    Parameters
+    ----------
+    hitc       : evm.HitCollection
+        Colection of hits
+    blob_pos1  : Tuple[float, float, float]
+        x, y, z tuple for the blob 1 position
+    blob_pos2  : Tuple[float, float, float]
+        x, y, z tuple for the blob 2 position
+    blob_radius: float
+        radius for the blob definition
+    energy_out : bool
+        if true, it also returns the blob energies
+
+    Returns
+    A tuple with the number of hits for both blobs and, if asked,
+    the energy of the blobs.
+    """
+    nhits1     = 0
+    eb1        = 0.
+    nhits2     = 0
+    eb2        = 0.
+
+    for hit in hitc.hits:
+        e_h   = hit.Ec if not np.isnan(hit.Ec) and hit.Ec>0 else 0.
```

Comment:
Hi @mmkekic , forgot to comment on this one. If a hit has nan or negative energy, I would be inclined NOT to count it as +1. I guess we exclude those hits when computing blob energies, and I would use the same set of hits to compute blob energy and blob number of hits. Otherwise things like (average hit energy) = (blob energy) / (blob nhits) are ill-defined.

In any case, I'd like to understand better if such hits can be real or are pure artifacts. What kind of physics situation creates  a hit, but one with nan or negative energy?

Reply:
I agree, I was simply mimicking @msorel procedure, where this protection was added... Did you have to do it because you found misbehaviour or just it was general protection just in case?

Reply:
The protection against nan energies is necessary, I think. The code crashed otherwise. We are doing things like
eb1    += e_h
here, which make no sense with nan e_h, I guess.